flutter/lib/src/sentry_flutter.dart

```diff
@@ -10,34 +9,36 @@ import 'package:sentry/sentry.dart';
 import 'default_integrations.dart';
 import 'file_system_transport.dart';
 import 'version.dart';
+import 'web_platform_checker.dart' if (dart.library.io) 'platform_checker.dart';
 
 /// Sentry Flutter SDK main entry point
 mixin SentryFlutter {
   static const _channel = MethodChannel('sentry_flutter');
 
   static Future<void> init(
-    OptionsConfiguration optionsConfiguration,
-    Function callback, {
+    OptionsConfiguration optionsConfiguration, {
+    AppRunner appRunner,
```

Comment:
on `Sentry.init` we use: [] arguments, so we don't need to pass the name.
why did we do a different approach here?

I think it'd be less verbose to just doing the same, `[ AppRunner appRunner ]` so a name is not required as this is most probably the most common scenario.

Reply:
we already had named arguments ( `packageLoader`and `isIOSChecker` ), we can't have both named and anonymous optional arguments. Do you prefer 3 not named arguments ?
```dart
static Future<void> init(
    OptionsConfiguration optionsConfiguration,
    Function callback, {
    PackageLoader packageLoader = _loadPackageInfo,
    iOSPlatformChecker iOSPlatformChecker = _iOSPlatformChecker,
  }) async {}
```

Reply:
I see, I didn't know this limitation.

wondering if we could clean up the PackageLoader and iOSPlatformChecker from the `init` method at all then.

on Android, we add all the necessary add ons to the SentryOptions and set default values to them (adding a public setter to overwrite on tests), so we could still mock and test.

My concern is, if we do a different way of initing the SDK, we'll need to change all the documentation to reflect the way of initing it.

We can start with the `[]` params to merge finish this task and refactor later as this won't be a breaking change anyway, those classes ar only used for testing anyway, wdyt?

Reply:
✅ I'm agree : as it's mainly for testing purpose, we totally can make them positional for now, and refactoring it later to remove them.

Reply:
create a task for this in the board, to not be forgotten